nexus_constructor/StatusConsumer.py

```diff
@@ -0,0 +1,111 @@
+from kafka import KafkaConsumer, TopicPartition
+import threading
+from copy import copy
+import time
+from kafka.errors import NoBrokersAvailable
+import json
+
+
+class StatusConsumer:
+    def __init__(self, broker, topic):
+        self.broker = broker
+        self.topic = topic
+        self.connected = False
+        self.filewriters = {}
+        self.files = {}
+        self.status_lock = threading.Lock()
+        self.run_thread = True
+
+        self.thread = threading.Thread(target=self.status_thread)
+        self.thread.start()
+
+    def __del__(self):
+        self.run_thread = False
+        self.thread.join()
+
+    def isConnected(self):
+        self.status_lock.acquire()
+        return_status = copy(self.connected)
+        self.status_lock.release()
+        return return_status
+
+    def getFilewriters(self):
+        self.status_lock.acquire()
+        return_filewriters = copy(self.filewriters)
+        self.status_lock.release()
+        return return_filewriters
+
+    def getFiles(self):
+        self.status_lock.acquire()
+        return_files = copy(self.files)
+        self.status_lock.release()
+        return return_files
+
+    def _setConnected(self, is_connected):
+        self.status_lock.acquire()
```

Comment:
maybe a bit nit-picky but could this be done with a context manager of some sorts? it is duplicated in most functions